core/src/main/java/io/questdb/cutlass/text/CairoTextWriter.java

```diff
@@ -254,8 +256,12 @@ private TableWriter openWriterAndOverrideImportTypes(
                         this.types.setQuick(i, BadDateAdapter.INSTANCE);
                         break;
                     case ColumnType.TIMESTAMP:
-                        logTypeError(i);
-                        this.types.setQuick(i, BadTimestampAdapter.INSTANCE);
+                        if (detectedType == ColumnType.DATE) {
+                            this.types.setQuick(i, dateToTimestampAdapter.of((DateAdapter) this.types.getQuick(i)));
```

Comment:
Table can have more than one timestamp and more than one date that is misplaced against target column type. A single instance of `dateToTimestampAdapter` can be overwritten.

There is ObjectPool that can produce multiple adaptors like that when needed.